### ISSUE TITLE
Edge occassionally shows a "set up/signin" modal after startup, causing testcafe to fail

### ISSUE THREAD

**Issue by christian-fischer** [DevExpress/testcafe#8149]
### What is your Scenario?

Using Edge with native automation enabled.

### What is the Current behavior?

Occassionally [1] immediately after testcafe starts Edge, a modal appears regarding some profile setup/signin (see screenshot).
This subsequently fails the testrun, as not even the `http://localhost:4300/browser/connect/*` url is loaded, due to the modal blocking the browser.

[1] sometimes as frequent as every second run, other times as rarely as every ~20th run

### What is the Expected behavior?

I expect no modal to appear and that Edge loads the page I want to test.

### What is the public URL of the test page? (attach your complete example)

Issue happens before the test page is even loaded. So for testing purposes using any page is fine, i.e. https://testcafe.io

### What is your TestCafe test code?

```
fixture('test');
test('edge', async (tc) => {
  await tc.navigateTo('https://testcafe.io');
  await tc.expect(true).ok();
});
```

### Your complete configuration file

```
// no specific configuration needed, fails with defaults
```

### Your complete test report

```
Live mode is enabled.
TestCafe now watches source files and reruns
the tests once the changes are saved.

You can use the following keys in the terminal:
'Ctrl+S' - stops the test run;
'Ctrl+R' - restarts the test run;
'Ctrl+W' - enables/disables watching files;
'Ctrl+C' - quits live mode and closes the browsers.


Watching the following files:
  .\edge.test.js

Stopping TestCafe live mode...

Make changes to the source files or press Ctrl+R to restart the test run.
ERROR Unable to open the "edge:" browser due to the following error:

TypeError: Cannot read properties of undefined (reading 'Fetch')
    at NativeAutomationRequestPipeline.start (.\node_modules\testcafe\src\native-automation\request-pipeline\index.ts:387:28)
    at NativeAutomation.start (.\node_modules\testcafe\src\native-automation\index.ts:46:29)
    at BrowserProviderPluginHost._setupNativeAutomation (.\node_modules\testcafe\src\browser\provider\built-in\dedicated\chrome\index.js:63:32)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at BrowserProviderPluginHost.openBrowser (.\node_modules\testcafe\src\browser\provider\built-in\dedicated\chrome\index.js:109:13)
    at BrowserProvider.openBrowser (.\node_modules\testcafe\src\browser\provider\index.ts:326:9)
    at BrowserConnection._runBrowser (.\node_modules\testcafe\src\browser\connection\index.ts:279:13)

Type "testcafe -h" for help.
```

### Screenshots

![testcafe-edge-issue](https://github.com/DevExpress/testcafe/assets/5974170/2eaaf8af-ff78-4ef7-90d6-2b63d9601961)

### Steps to Reproduce

1. run `testcafe edge edge.test.js -L`
2. observe the modal interfering with test execution

### TestCafe version

3.2.0

### Node.js version

18.12.0

### Command-line arguments

`testcafe edge edge.test.js`

### Browser name(s) and version(s)

Edge 122.0.2365.52

### Platform(s) and version(s)

Windows 11 23H2

**Comment by PavelMor25** [comment]
Hello @christian-fischer,

We can't reproduce the issue on our machines ( windows 10-11 / nodeJS 18-20 / Edge 122.0.2365.52-59 / testcafe 3.2 - 3.5).
 
Could you please try to reproduce the issue on other machines?

Perhaps, you can specify details that can help us reproduce the problematic behavior.

**Comment by christian-fischer** [comment]
Hi @PavelMor25 
when I tried reproducing this, it only happened occasionally - so I had to execute the command several times to get one failing run due to the described issue.
I'm not sure how edge determines when to show this modal dialog or not - maybe there are pre-requisites regarding the windows-account?

As testcafe controls the profile in native automation, maybe it's possible to set some edge policy? The [HideFirstRunExperience`](https://learn.microsoft.com/en-us/DeployEdge/microsoft-edge-policies#hidefirstrunexperience) policy sounds promising in this regard to suppress this modal dialog.
